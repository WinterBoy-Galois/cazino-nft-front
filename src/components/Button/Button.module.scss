@import '../../styles/shared';

.button {
  width: 100%;
  border-radius: 8px;
  color: $white;
  text-transform: uppercase;
  font-size: 20px;
  background-color: $button-primary;
  border: 0;
  padding: 0.3rem;

  &--small {
    padding: 5px;

    @include media-breakpoint-up(lg) {
      padding: 7px;
    }
  }

  &--medium {
    padding: 9px;

    @include media-breakpoint-up(lg) {
      padding: 14px;
    }
  }

  &--large {
    padding: 9px;

    @include media-breakpoint-up(lg) {
      padding: 18px;
    }
  }

  &:disabled {
    opacity: 0.5;
    cursor: not-allowed;
  }

  &:hover,
  &:focus {
    background-color: lighten($button-primary, 10%);
  }
}

