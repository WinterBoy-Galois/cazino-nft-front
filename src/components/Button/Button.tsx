import React from 'react';
import styles from './Button.module.scss';
import { ButtonSize } from './lib/size';

interface IProps {
  type?: 'submit' | 'reset' | 'button';
  className?: string;
  onClick?: () => void;
  size?: ButtonSize;
  disabled?: boolean;
}

const Button: React.FC<IProps> = props => {
  const { type = 'button', className = '', size = ButtonSize.SMALL } = props;
  return (
    <button
      {...props}
      className={`${className} ${styles.button} ${styles[`button--${size.toString()}`]}`}
      type={type}
    />
  );
};

export default Button;
