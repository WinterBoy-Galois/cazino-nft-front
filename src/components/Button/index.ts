export { default } from './Button';
export { ButtonSize } from './lib/size';
