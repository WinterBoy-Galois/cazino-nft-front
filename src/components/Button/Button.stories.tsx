import React from 'react';
import { storiesOf } from '@storybook/react';
import Button from '.';
import { action } from '@storybook/addon-actions';
import { withKnobs, text, select } from '@storybook/addon-knobs';
import { ButtonSize } from './lib/size';

storiesOf('Components/buttons/Button', module)
  .addDecorator(withKnobs)
  .addDecorator(storyFn => <div style={{ padding: '1rem', maxWidth: '200px' }}>{storyFn()}</div>)
  .add('default', () => (
    <Button
      onClick={action('button click')}
      size={select(
        'Size',
        {
          small: ButtonSize.SMALL,
          medium: ButtonSize.MEDIUM,
          large: ButtonSize.LARGE,
        },
        ButtonSize.SMALL
      )}
    >
      {text('Text', 'Click me')}
    </Button>
  ));
