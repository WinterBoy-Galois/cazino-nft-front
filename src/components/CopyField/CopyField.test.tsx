import React from 'react';
import { render } from '@testing-library/react';
import Button from '.';

describe('Button', () => {
  it('should match snapshot', () => {
    // Arrange

    // Act
    const container = render(<Button>Test</Button>);

    // Assert
    expect(container).toMatchSnapshot();
  });
});
