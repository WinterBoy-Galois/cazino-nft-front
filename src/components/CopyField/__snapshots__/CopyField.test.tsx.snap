// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button should match snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class=" container"
      >
        <div
          class="label"
        />
        <div
          class="value"
        />
        <div
          class="copy"
        >
          <svg
            class="default"
            viewBox="0 0 12.14 13.88"
            xmlns="http://www.w3.org/2000/svg"
          >
            <g
              data-name="Layer 2"
            >
              <g
                data-name="Layer 1"
              >
                <path
                  d="M8.67 12.14v1.09a.66.66 0 01-.65.65H.65a.66.66 0 01-.65-.65v-10a.65.65 0 01.65-.63h2v8a1.52 1.52 0 001.52 1.52zm0-9.32V0H4.12a.65.65 0 00-.65.65v10a.65.65 0 00.65.65h7.37a.65.65 0 00.65-.65V3.47H9.32a.65.65 0 01-.65-.65zM12 2L10.16.19A.66.66 0 009.7 0h-.16v2.6h2.6v-.16A.66.66 0 0012 2z"
                  data-name="Icon awesome-copy"
                />
              </g>
            </g>
          </svg>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class=" container"
    >
      <div
        class="label"
      />
      <div
        class="value"
      />
      <div
        class="copy"
      >
        <svg
          class="default"
          viewBox="0 0 12.14 13.88"
          xmlns="http://www.w3.org/2000/svg"
        >
          <g
            data-name="Layer 2"
          >
            <g
              data-name="Layer 1"
            >
              <path
                d="M8.67 12.14v1.09a.66.66 0 01-.65.65H.65a.66.66 0 01-.65-.65v-10a.65.65 0 01.65-.63h2v8a1.52 1.52 0 001.52 1.52zm0-9.32V0H4.12a.65.65 0 00-.65.65v10a.65.65 0 00.65.65h7.37a.65.65 0 00.65-.65V3.47H9.32a.65.65 0 01-.65-.65zM12 2L10.16.19A.66.66 0 009.7 0h-.16v2.6h2.6v-.16A.66.66 0 0012 2z"
                data-name="Icon awesome-copy"
              />
            </g>
          </g>
        </svg>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
